planetscale/certs.go

```diff
@@ -57,10 +59,16 @@ func (c *certificatesService) Create(ctx context.Context, r *CreateCertificateRe
 		CommonName: cn,
 	}
 
+	switch r.PrivateKey.(type) {
+	case *rsa.PrivateKey:
+	case *ecdsa.PrivateKey:
+	default:
+		return nil, fmt.Errorf("unsupported key type: %T, only supports ECDSA and RSA private keys", r.PrivateKey)
+	}
+
 	template := x509.CertificateRequest{
-		Version:            1,
-		Subject:            subj,
-		SignatureAlgorithm: x509.SHA256WithRSA,
```

Comment:
The default signature algorithms are the correct one for both RSA & ECDSA keys so no need to overwrite it.